refactor: Adds GSDP info, cleans Dokku docs (#356)
codeRIT/hackathon-manager: docs/deployment-dokku.md

>These docs assume you already have a virtual machine with [Dokku](http://dokku.viewdocs.io/dokku/) running on it, and can SSH into the VM. DNS should be set up as well, but isn't required for bare minimum functionality.
>
>If you need a VM, check out [DigitalOcean](https://m.do.co/c/b5ee103e23c3) or [Linode](https://www.linode.com/?r=e90a6fb2a6999fb4ec7b60b1add3e288f97954bf) and the [Dokku docs](http://dokku.viewdocs.io/dokku/) to get started.
>

> >**Student Developer?** The [GitHub Student Developer Pack](https://education.github.com/pack?sort=popularity&tag=Cloud) has several discounts for cloud hosting. 

Reviewer: Maybe we can plug `cloud hosting, Ruby tutorials, and more!`

## Setting up a new deployment

> Below are steps & notes to deploy a new HackathonManager instance on Dokku. To update an existing Dokku deployment, check out our [updating docs](updating-hm.html).

Reviewer: ```suggestion
Below are steps to deploy a new HackathonManager instance on Dokku. To update an existing Dokku deployment, check out our [updating docs](updating-hm.html).
```

> If you have any questions at all, please don't hesitate to reach out to the [codeRIT Engineering Team](mailto:engineering@coderit.org)! This doc is very much a work in progress but we want to keep it as up to date as possible.

Reviewer: ```suggestion
If you have any questions, please don't hesitate to reach out to the [codeRIT Engineering Team](mailto:engineering@coderit.org)! This doc is very much a work in progress but we want to keep it as up to date as possible.
```

Reviewer: Slightly out of scope but while we're here...

## Dokku Setup

### Plugins

Currently used and required Dokku plugins (other than the defaults):

- [MySQL](https://github.com/dokku/dokku-mysql) (data storage)
- [Redis](https://github.com/dokku/dokku-redis) (background jobs + caching)
- [dokku-letsencrypt](https://github.com/dokku/dokku-letsencrypt) (Optional: free, automated SSL certificates)

### Setup Steps

> **We'll be using `hm` as the app name in these steps,** as well as sharing the same `hm` name for both the app, database, and redis name. You're free to use another names.

Reviewer: ```suggestion
**We'll be using `hm` as the app name in these steps,** as well as sharing the same `hm` name for both the app, database, and redis name. You're free to use another name for your hackathon.
```

Author: I would make it `You're free to use another name for your app (or HM instance.)` Hackathon implies this has direct impact on the naming of the hackathon throughout HM.

Reviewer: Let's do that, and this PR is ready :shipit:

Author: Decided it would probably be best to go with the original but replace "another" for "other." By saying what I initially suggested, it implies that only the app was adjustable when in fact, all the names are adjustable.
